src/user-guide/running-workflows/reflow.rst

```diff
@@ -1,86 +1,227 @@
-a corrupted ancillary file resulted in bad products at cycle 5.
 
-To rectify this we could fix the corrupted file and trigger a new flow
-(a reflow) from ``5/post``::
+.. _triggering-flows:
 
-   cylc trigger --reflow <workflow-id>//5/post
+Triggering & Flows
+------------------
 
-The new flow will regenerate and republish cycle 5 products before naturally
-coming to a halt, because the triggered tasks do not lead on to the next cycle
-and the rest of the graph.
+By default, manual triggering (with ``cylc trigger`` or the UI) starts a new
+:term:`front of activity<flow front>` in current flows.
+But it can also start new flows and trigger flow-independent single tasks.
 
-Meanwhile, the original flow will carry on unaffected, from cycle 8.
+In the diagrams below, the grey nodes represent tasks in flow ``1``, and
+the blue ones stem from a single manual triggering event.
 
-Flow Merging
-------------
+Triggering in Current Flows
+   ``cylc trigger [--wait] ID``
+
+   This is the default trigger action: trigger the task and give it all current
+   active flow numbers. Subsequently, each of those flows will consider this
+   task to have run already.
+
+   **Ahead of active flows** this starts a new front of activity for the active
+   flows, which by default can continue on without waiting for catch up.
+
+   .. image:: ../../img/same-flow-n.png
+
+   With ``--wait``, the result is the same except that any action downstream of 
+   the triggered task is delayed until the first flow catches it.
+
+   **Behind active flows** the triggered task itself will re-run, then activity
+   will cease if any of the original flows already traversed that part of the
+   graph.
+
+Triggering in Specific Flows
+   ``cylc trigger --flow=1,2 ID``
+
+   This triggers the task with flow numbers ``1`` and ``2``.
+
+   The result is like the default above, except that tasks in the new front
+   belong only to the specified flow(s), regardless of which flows are
+   :term:`active` at triggering time.
+
+Triggering a New Flow
+   ``cylc trigger --flow=new ID``
+
+   This triggers the task with a new, incremented flow number.
+
+   The new flow will re-run tasks that already ran in previous flows.
+
+   .. image:: ../../img/new-flow-n.png
+
+
+Triggering a Flow-Independent Single Task
+   ``cylc trigger --flow=none ID``
+
+   This triggers a task with no flow numbers.
+
+   It will not spawn children, and other flows that come by will re-run it.
+
+   .. image:: ../../img/no-flow-n.png
+
+Special Case: Triggering ``n=0`` Tasks
+   Tasks in the ``n=0`` window are :term:`active`, :term:`active-waiting`, or
+   :term:`incomplete`. Their flow membership is already determined - that of
+   the parent tasks that spawned them.
+
+   - Triggering an :term:`active task` has no effect (it is already triggered).
+   - Triggering an :term:`active-waiting task` runs it immediately in the same flow.
+   - Triggering an :term:`incomplete task` re-runs it immediately in the same flow.
```

Comment:
I understand from @oliver-sanders this isn't quite right.
Triggering an active-waiting task that's waiting on an internal queue runs it immediately.
In all other cases the trigger queues the task so it may require a second trigger to run immediately.